apps/microtvm/zephyr/template_project/src/host_driven/main.c

```diff
@@ -64,8 +67,89 @@ static size_t g_num_bytes_requested = 0;
 static size_t g_num_bytes_written = 0;
 static size_t g_num_bytes_in_rx_buffer = 0;
 
+#ifdef FVP
```

Comment:
what if we break this out into a separate file and modify CMakeLists.txt to include it (we can still e.g.
```
#ifdef FVP
#include "semihost.h"
#endif
```

Reply:
done